kstreams/engine.py

```diff
@@ -188,15 +189,13 @@ def stream(
         name: Optional[str] = None,
         deserializer: Optional[Deserializer] = None,
         **kwargs,
-    ) -> Callable[[Callable[[Stream], None]], Stream]:
-        def decorator(func: Callable[[Stream], None]) -> Stream:
-            stream = self._create_stream(
-                topics,
-                func=func,
-                name=name,
-                deserializer=deserializer,
-                **kwargs,
+    ) -> Callable[[Func], Stream]:
+        def decorator(func: Func) -> Stream:
+            _stream = stream(topics, name=name, deserializer=deserializer, **kwargs)(
```

Comment:
I think there is no need to make stream private.

Reply:
it would replace the `stream` function imported

Reply:
ah, this is why. It confused me because I thought the idea was to make the `stream` private. It could be ok but It might confuse others as well.

Reply:
I will rename it